modules/chat/client/controllers/chat.client.controller.js

```diff
@@ -1,43 +1,55 @@
-'use strict';
+(function () {
+  'use strict';
 
-// Create the 'chat' controller
-angular.module('chat').controller('ChatController', ['$scope', '$location', 'Authentication', 'Socket',
-  function ($scope, $location, Authentication, Socket) {
-    // Create a messages array
-    $scope.messages = [];
+  angular
+    .module('chat')
+    .controller('ChatController', ChatController);
 
-    // If user is not signed in then redirect back home
-    if (!Authentication.user) {
-      $location.path('/');
-    }
+  ChatController.$inject = ['$scope', '$state', 'Authentication', 'Socket'];
 
-    // Make sure the Socket is connected
-    if (!Socket.socket) {
-      Socket.connect();
-    }
+  function ChatController($scope, $state, Authentication, Socket) {
+    var vm = this;
```

Comment:
I think "vm" is best. If you start getting into more specific names (i.e. chat), then you'll end up dancing around that name, when naming other more ambiguous members/properties of a controller. To me it's very clear what "vm" is when looking at the code; it's very accurately named, IMO.